src/cpp/library/CUDA/cuda_runtime_api.h

```diff
@@ -50,6 +51,37 @@ cudaError_t cudaMemcpy(T1 dst, T2 src, size_t count, enum cudaMemcpyKind kind)
   return __cudaMemcpy(dst, src, count, kind);
 }
 
+cudaError_t cudaMalloc(void **devPtr, size_t size)
+{
+  cudaError_t tmp;
+  //pre-conditions
+  __ESBMC_assert(size > 0, "Size to be allocated may not be less than zero");
+  *devPtr = malloc(size);
+
+  if(*devPtr == NULL)
+  {
+    tmp = CUDA_ERROR_OUT_OF_MEMORY;
+    exit(1);
```

Comment:
According to [the docs](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__MEMORY.html#group__CUDART__MEMORY_1g37d37965bfb4803b6d4e59ff26856356):
> cudaMalloc() returns cudaErrorMemoryAllocation in case of failure.

Is there a reason to call `exit(1)` here?

Reply:
I will fix this issue in this PR: https://github.com/esbmc/esbmc/pull/1245/files